storage/client_test.go

```diff
@@ -1342,6 +1347,47 @@ func TestObjectConditionsEmulated(t *testing.T) {
 	})
 }
 
+// Test that RetryNever prevents any retries from happening in both transports.
+func TestRetryNeverEmulated(t *testing.T) {
+	transportClientTest(t, func(t *testing.T, project, bucket string, client storageClient) {
+		ctx := context.Background()
+
+		attrs, err := client.CreateBucket(ctx, project, bucket, &BucketAttrs{}, nil)
+		if err != nil {
+			t.Fatalf("creating bucket: %v", err)
+		}
+
+		// Need the HTTP hostname to set up a retry test, as well as knowledge of
+		// underlying transport to specify instructions.
+		host := os.Getenv("STORAGE_EMULATOR_HOST")
+		endpoint, err := url.Parse(host)
+		if err != nil {
+			t.Fatalf("parsing endpoint: %v", err)
+		}
+		var transport string
+		if _, ok := client.(*httpStorageClient); ok {
+			transport = "http"
+		} else {
+			transport = "grpc"
+		}
+
+		et := emulatorTest{T: t, name: "testRetryNever", resources: resources{},
+			host: endpoint}
+		et.create(map[string][]string{"storage.buckets.get": {"return-503"}}, transport)
+		ctx = callctx.SetHeaders(ctx, "x-retry-test-id", et.id)
+		_, err = client.GetBucket(ctx, attrs.Name, nil, withRetryConfig(&retryConfig{policy: RetryNever}))
+
+		var ae *apierror.APIError
+		if errors.As(err, &ae) {
+			if ae.GRPCStatus().Code() == codes.Unavailable || ae.HTTPCode() == 503 {
+				return
```

Comment:
We could probably just invert the condition and error here instead of returning

Reply:
Sure, I'll change that.